### PR TITLE
Updated auth rule

### PR DESCRIPTION
**Pull request by matt-goldman**
Closes https://github.com/SSWConsulting/SSW.Rules.Content/issues/22

### CODE REVIEW

**Review by chrisschultzssw — APPROVED** [review]
LGTM! Is this the first rule to mention Entra ID?

**Comment by matt-goldman** [comment]
> LGTM! Is this the first rule to mention Entra ID?

I believe so!

**Comment by bradystroud** [comment]
@BrookJeynes Image check failed - please take a look at this

**Review by GordonBeeming — APPROVED** [review]
LGTM 🚢

**Review by Hona — APPROVED** [review]
LGTM - I love the decision tree

**Comment by GordonBeeming** [comment]
The broken image could be related to a bug, I've fixed it now, logs are gone from this run to check if that was the reason

https://github.com/SSWConsulting/SSW.Rules.Content/pull/6792

**Comment by matt-goldman** [comment]
> The broken image check could be related to a bug, I've fixed it now, logs are gone from this run to check if that was the reason
> 
> #6792

I just ran that check again and got the same result

File: rules/choosing-authentication/rule.md [hunk 1]
```diff
@@ -31,80 +31,64 @@ Choosing the right [authentication and authorization](https://www.youtube.com/wa
...
-      - [OIDC](https://openid.net/connect)?
-3. **MFA** - Do you need to support   
-      - [MFA](https://en.wikipedia.org/wiki/Multi-factor_authentication)?
-4. **Scope** - Do you need to share the identity across multiple applications?
-5. **Volume** - Do you have an estimate for how many users you need to support?
-6. **Complexity** - do you need custom logic to run as part of your authentication process?
+* **Scope** - Is it an enterprise application for internal users or a consumer application for external use?
+* **Scope** - Do you need to share the identity across multiple applications?
+* **Volume** - Do you have an estimate for how many users you need to support?
+* **Complexity** - do you need custom logic to run as part of your authentication process?
```
Comment (drwharris) [inline rules/choosing-authentication/rule.md:37]: I understand whats being asked here but most people wont know what is going to end up being custom logic and personally its the one thing that will trip people up. If you say no here and turns out you do, you are in trouble, so I think its worth while teasing this out a bit.

Complexity - Do you need additional logic run in your authentication process? some examples might be: 

calling out to other systems to add additional claims,
....

I would steer clear of "Custom" everywhere, IMO decision makers hate it.

Comment (danielmackay) [reply rules/choosing-authentication/rule.md:37]: Agree with @drwharris.  I would like to see some real world examples as to what 'custom logic' means.

File: rules/choosing-authentication/rule.md [hunk 2]
```diff
@@ -140,28 +127,58 @@ Advantages:
 * Allows you to define custom authentication logic
```
Comment (danielmackay) [inline rules/choosing-authentication/rule.md:127]: Same comment as above.  Can you provide some more details around what 'custom logic means'?

File: rules/choosing-authentication/rule.md [hunk 3]
```diff
@@ -140,28 +127,58 @@ Advantages:
...
 :::
 
-* Steep learning curve
-* Requires additional setup
-* Requires additional skillset
-* Requires additional management
+* Learning curve
+* Requires additional hosting resources
+* Requires additional skill set
+* Requires ongoing maintenance
```
Comment (danielmackay) [inline rules/choosing-authentication/rule.md:139]: Is it worth mentioning the cost here (as some of the other options are free - at least at low usage)?

**Review by danielmackay — APPROVED** [review]
Approved with comments.

**Comment by GordonBeeming** [comment]
> I just ran that check again and got the same result

@matt-goldman taking a look now 🙂

File: rules/choosing-authentication/rule.md (hunk 1 above)
Comment (matt-goldman) [reply rules/choosing-authentication/rule.md]: Good point. Done.

File: rules/choosing-authentication/rule.md (hunk 2 above)
Comment (matt-goldman) [reply rules/choosing-authentication/rule.md]: Agreed - done.

File: rules/choosing-authentication/rule.md (hunk 3 above)
Comment (matt-goldman) [reply rules/choosing-authentication/rule.md:151]: Good point - added

**Pull request merged**